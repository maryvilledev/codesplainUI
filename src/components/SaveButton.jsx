import React, { PropTypes } from 'react';
import { browserHistory } from 'react-router';
import IconButton from 'material-ui/IconButton';
import Save from 'material-ui/svg-icons/content/save';
import Snackbar from 'material-ui/Snackbar';

export default class SaveButton extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      open: false,
      message: '',
    };
    this.handleTouchTap = this.handleTouchTap.bind(this);
    this.handleRequestClose = this.handleRequestClose.bind(this);
  }

  handleTouchTap() {
    this.setState({
      open: true,
      message: '',
    });
    this.props.onSaveClick()
      .then((res) => {
        const id = res.data.id;
        browserHistory.push(`/${id}`);
        this.setState({ message: 'Codesplaination saved!' });
      }, (err) => {
        console.error(err);
        this.setState({ message: 'Error saving snippet data' });
      });
  }

  handleRequestClose() {
    this.setState({
      open: false,
    });
  }

  render() {
    return (
      <div>
        <IconButton
          onTouchTap={this.handleTouchTap}
          tooltip="Save snippet"
        >
          <Save />
        </IconButton>
        <Snackbar
          open={this.state.open}
          message={this.state.message}
          autoHideDuration={2000}
          onRequestClose={this.handleRequestClose}
        />
      </div>
    );
  }
}

SaveButton.propTypes = {
  onSaveClick: PropTypes.func.isRequired,
};
