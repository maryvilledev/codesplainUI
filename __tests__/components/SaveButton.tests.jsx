import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import getMuiTheme from 'material-ui/styles/getMuiTheme';

import SaveButton from '../../src/components/buttons/SaveButton';

describe('<SaveButton />', () => {
  const muiTheme = getMuiTheme();
  const shallowWithContext = (node) => shallow(node, { context: { muiTheme } });

  it('matches snapshot', () => {
    const wrapper = shallowWithContext(
      <SaveButton
        onSaveClick={jest.fn()}
      />
    );
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  })
})
