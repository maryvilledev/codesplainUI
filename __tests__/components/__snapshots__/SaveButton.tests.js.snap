exports[`<SaveButton /> matches snapshot 1`] = `
<div>
  <IconButton
    disableTouchRipple={false}
    disabled={false}
    iconStyle={Object {}}
    onTouchTap={[Function]}
    tooltip="Save snippet"
    tooltipPosition="bottom-center"
    touch={false}>
    <ContentSave />
  </IconButton>
  <Snackbar
    autoHideDuration={2000}
    message=""
    onRequestClose={[Function]}
    open={false} />
</div>
`;
